### ISSUE TITLE
Expose RouteProgress Information in onProgressChange Event

### ISSUE THREAD

**Issue by devguy22** [homeeondemand/react-native-mapbox-navigation#8]
Thank you for your library. We were wondering if you could expose some of values of the RouteProgress in the navigationViewController didUpdate method? We need at least the primitives: distanceTraveled, durationRemaining, fractionTraveled, and distanceRemaining. These are the iOS names for those values, and I'm unfamiliar with what they would be on the Android side.

These would be exposed as extra parameters to the onProgressChange callback, alongside latitude and longitude. Let me know if I can assist (at least with the iOS side).

**Comment by rossmartin** [comment]
Hey you're welcome. Yeah I'll update `onProgressChange` to include the extra data that is available. The challenge is keeping it consistent between platforms but I'll try to do that. I might have time to tackle this later tonight or tomorrow night - I'll post back here when it's released.

**Comment by rossmartin** [comment]
I added a new `onRouteProgressChange` prop that gets called with distanceTraveled, durationRemaining, fractionTraveled, and distanceRemaining. Note that the `onProgressChange` has been renamed to `onLocationChange`. I published a new major version at `1.0.0` with the changes.

The delegate event on iOS includes both "location" and "route progress" together but on android it has two separate "listeners". This is why I had to separate the props to keep it consistent.
